package/run.ps1

```diff
@@ -31,28 +70,111 @@ function New-Directory {
     }
 }
 
-function Get-StringHash {
+function Get-CurrentEnvHash {
     [CmdletBinding()]
     param (
         [Parameter()]
         [string]
-        $Value
+        $rke2ServiceName
     )
-    $stringAsStream = [System.IO.MemoryStream]::new()
-    $writer = [System.IO.StreamWriter]::new($stringAsStream)
-    $writer.write($Value)
-    $writer.Flush()
-    $stringAsStream.Position = 0
-    return (Get-FileHash -InputStream $stringAsStream -Algorithm SHA256).Hash.ToLower()
+    try
+    {
+        $currentEnv = Get-ItemProperty HKLM:SYSTEM\CurrentControlSet\Services\$rke2ServiceName -ErrorAction SilentlyContinue
+        if ($null -eq $currentEnv)
+        {
+            # either no environment variables have been set, or rke2 hasn't been installed
+            return ""
+        }
+        $envString = ($currentEnv).Environment | Out-String
+        return Get-StringHash -Value ($envString)
+    } catch {
+        return ""
+    }
+}
+
+function Determine-Upgrading {
+    $CURRENT_RKE2_VERSION = ""
+    try
+    {
+        $RKE2_VERSION_OUTPUT = $(c:\usr\local\bin\rke2.exe --version)
```

Comment:
When should drive letters be capitalized?

Reply:
it doesn't matter. I think canonically they are upper case but you will never get an error if you specify it in lowercase, it'd only be a problem if you were comparing strings or something.

Reply:
Yep, powershell can handle both `c:/` and `C:/`

Reply:
c:/ and c:\ are also treated identically 🤷🏻

Reply:
I was moreso asking from a coding standards POV, anyway resolving.